google/cloud/pubsublite/internal/batching_settings.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_BATCHING_SETTINGS_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_BATCHING_SETTINGS_H
+
+#include "google/cloud/version.h"
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace pubsublite_internal {
+
+class BatchingSettings {
+ public:
+  BatchingSettings(size_t byte_limit, size_t message_limit)
```

Comment:
This class should be structured like PublisherOptions http://google3/third_party/cloud_cpp/google/cloud/pubsub/publisher_options.h;l=61;rcl=418678942, be called "BatchingOptions", and have the same batching-related setters and getters. It should also be in pubsublite/ not pubsublite/internal

I wouldn't take the approach of "if you mess up, oh well" taken by those options though- I would instead clamp the values in the setters to between 0 and the maximum allowed (1000 for count, 3.5 MiB for bytes). Alternatively, you could perform the clamping in the PartitionPublisher class.